warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,278 @@
+//
+//   Copyright 2021  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.ext.http;
+
+import io.warp10.WarpConfig;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WebAccessController;
+import io.warp10.script.formatted.FormattedWarpScriptFunction;
+import io.warp10.standalone.StandaloneWebCallService;
+import io.warp10.warp.sdk.Capabilities;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Apply an HTTP method over an url
+ */
+public class HTTP extends FormattedWarpScriptFunction {
```

Comment:
Use extends NamedWarpScriptFunction implements WarpScriptStackFunction for clarity. The FormattedWarpScriptFunction approach does cripple the overall readability and may be removed in future versions of Warp 10.

Reply:
I think after knowing how to use the FormattedWarpScriptFunction class it is the opposite. I find it handy to declare functions with multiple arguments to retrieve from the stack and handle defaults, while also hiding the stack syntax particularity. Also it is convenient to generate documentation files.

Maybe the logic of this class can be transferred to a stack method: stack.retrieveArguments(args)
 And documentation generation to a function of args, output and function.